### PR TITLE
Simplify IAssemblyReference

### PR DESCRIPTION
**Pull request by tmat**
The only property that's needed is Identity returning AssemblyIdentity.

### CODE REVIEW

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Portable/Emitter/Model/AssemblyReference.cs [hunk 1]
```diff
@@ -35,6 +35,7 @@ void Cci.IReference.Dispatch(Cci.MetadataVisitor visitor)
             visitor.Visit(this);
         }
 
+#if F
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Emitter/Model/AssemblyReference.cs]: Remove the block?

Comment (tmat) [inline src/Compilers/CSharp/Portable/Emitter/Model/AssemblyReference.cs]: Yeah, forgot that one :)

File: src/Compilers/CSharp/Portable/Emitter/Model/AssemblyReference.cs [hunk 2]
```diff
@@ -94,6 +92,6 @@ Cci.IDefinition Cci.IReference.AsDefinition(CodeAnalysis.Emit.EmitContext contex
             return null;
         }
 
-        #endregion
+#endregion
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Emitter/Model/AssemblyReference.cs]: Is this intentional change? The directives are not aligned now.

Comment (tmat) [inline src/Compilers/CSharp/Portable/Emitter/Model/AssemblyReference.cs]: Fixed.

**Comment by AlekseyTs** [comment]
LGTM

File: src/Compilers/Core/Portable/Emit/ErrorType.cs [hunk 3]
```diff
@@ -174,53 +174,23 @@ string Cci.INamedEntity.Name
...
+        private sealed class ErrorAssembly : Cci.IAssemblyReference
         {
             public static readonly ErrorAssembly Singleton = new ErrorAssembly();
+            
             /// <summary>
             /// For the name we will use a word "Error" followed by a guid, generated on the spot.
             /// </summary>
-            private static readonly string s_name = "Error" + Guid.NewGuid().ToString("B");
+            private static readonly AssemblyIdentity s_identity = new AssemblyIdentity(
+                name: "Error" + Guid.NewGuid().ToString("B"),
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/Emit/ErrorType.cs]: Is this name ever surfaced to the user?  Curious about the impact of a random name on determinism (errors shuold ideally be deterministic as well).

Comment (tmat) [inline src/Compilers/Core/Portable/Emit/ErrorType.cs]: Not sure. The comment at the use site of this singleton says

```
// Substitute error types with a special singleton object.
// Unreported bad types can come through NoPia embedding, for example.
```

@AlekseyTs ?

Comment (tmat) [inline src/Compilers/Core/Portable/Emit/ErrorType.cs]: In any case this PR doesn't change the behavior. We can follow up later on removing the non-determinism. 

Can I get one more +1 for the change please?

**Comment by cston** [comment]
LGTM

File: src/Compilers/Core/Portable/Emit/ErrorType.cs (hunk 3 above)
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Emit/ErrorType.cs]: It is very unlikely for this name to surface to a user.

**Pull request merged**